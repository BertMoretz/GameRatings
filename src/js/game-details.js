import React from 'react'
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Grid from '@material-ui/core/Grid';
import axios from 'axios';
import CircularProgress from '@material-ui/core/CircularProgress';

import styles from "../styles.css"

export class GameDetails extends React.Component {

    state = { game: void 0 }

    loadCharacterDetails = (gameID) => {
        axios
            .get(`https://cors-anywhere.herokuapp.com/http://api-v3.igdb.com/games/${gameID}?fields=name,cover.*,artworks.*,age_ratings.*,created_at,game_engines.*,platforms.*,screenshots.*,genres.*,summary,aggregated_rating&&expand=cover,artworks,genres,screenshots,platforms,game_engines,age_ratings`, {headers: {
              "user-key": "861c079a35348acf2360c08a2efc2e90"
            }})
            .then(response => {
              console.log('Axios returned', response.data)
              this.setState({
                game: response.data[0]
              })
            });
    }

    componentDidMount() {
        if (this.props.match.params) {
            this.loadCharacterDetails(this.props.match.params.id)
        }
    }


    render() {

        const { game } = this.state

        if(!game) {
          return <div className={styles.loading}> <CircularProgress /> </div>
        }

        return (
            <Grid container justify="center" alignItems="center">
                <Grid item xs={12} md={12} lg={12} justify="center">
                    <Card className={styles.detailcard}>
                        <CardMedia
                            className={styles.detailedcover}
                            image={"https://images.igdb.com/igdb/image/upload/t_screenshot_huge/" + game.screenshots[0].image_id  + ".jpg"}
                            title="Cover"
                        />
                        <CardContent>
                            <Typography gutterBottom variant="h5" component="h2">
                                {game.name}
                            </Typography>
                            <Typography>
                                Origin: {game.summary}
                            </Typography>

                            <hr/>

                            <Table className={''}>
                                <TableBody>
                                    <TableRow>
                                        <TableCell component="th" scope="row">genre</TableCell>
                                        <TableCell align="right">{game.genres[0].name}</TableCell>
                                    </TableRow>

                                    <TableRow>
                                        <TableCell component="th" scope="row">Release</TableCell>
                                        <TableCell align="right">{game.created_at}</TableCell>
                                    </TableRow>

                                    <TableRow>
                                        <TableCell component="th" scope="row">Rating</TableCell>
                                        <TableCell align="right">{game.aggregated_rating}</TableCell>
                                    </TableRow>
                                </TableBody>
                            </Table>

                        </CardContent>
                    </Card>
                </Grid>
            </Grid>
        )
    }
}
