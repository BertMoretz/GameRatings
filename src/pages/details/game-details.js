import React from 'react'
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Grid from '@material-ui/core/Grid';
import GridList from '@material-ui/core/GridList';
import GridListTile from '@material-ui/core/GridListTile';
import GridListTileBar from '@material-ui/core/GridListTileBar';
import Paper from '@material-ui/core/Paper';
import axios from 'axios';
import CircularProgress from '@material-ui/core/CircularProgress';

import styles from "./details.css"
import back from '../../imgs/emptyback.png'

export class GameDetails extends React.Component {

    state = { game: void 0 }

    loadCharacterDetails = (gameID) => {
        axios
            .get(`https://cors-anywhere.herokuapp.com/http://api-v3.igdb.com/games/${gameID}?fields=name,cover.*,artworks.*,created_at,game_engines.*,platforms.*,screenshots.*,genres.*,summary,aggregated_rating,involved_companies.company.*&&expand=cover,artworks,genres,screenshots,platforms,game_engines,involved_companies, involved_companies.company`, {headers: {
              "user-key": "bf64f9cdc5dcd51330d432e658773047"
            }})
            .then(response => {
              console.log('Axios returned', response.data)
              this.setState({
                game: response.data[0]
              })
            });
    }

    componentDidMount() {
        if (this.props.match.params) {
            this.loadCharacterDetails(this.props.match.params.id)
        }
    }

    componentWillReceiveProps(nextProps) {
      if (nextProps.match.params.id != this.props.match.params.id) {
          this.loadCharacterDetails(nextProps.match.params.id);
      }
    }

    render() {

        const { game } = this.state

        if(!game) {
          return <div className={styles.loading}> <CircularProgress /> </div>
        }

        return (
            <Grid container justify="center" alignItems="center">
                <Grid item xs={12} md={12} lg={12} justify="center">
                  <div className={styles.hey}>
                    <img src={game.screenshots ? "https://images.igdb.com/igdb/image/upload/t_screenshot_huge/" + game.screenshots[0].image_id  + ".jpg" : back} className={styles.back}/>
                  </div>
                  <div className={styles.gamecover}>
                    <Grid container justify="center" alignItems="center">
                      <Grid item xs={12} md={4} lg={3} justify="center">
                        <img src={game.cover ? "https://images.igdb.com/igdb/image/upload/t_cover_big/" + game.cover.image_id + ".jpg": back} alt={game.name} className={styles.elem} />
                      </Grid>
                      <Grid item xs={12} sm container>
                        <Grid item xs={12} sm container direction="column" spacing={40}>
                          <Grid item >
                            <div className={styles.gamename}> {game.name} </div>
                            <div className={styles.gamegenre}> {game.genres ? game.genres[0].name : "Video Game"} </div>
                            <br/>
                            {
                              game.involved_companies ?
                              <div className={styles.gamecomp}> Developed by {game.involved_companies[1] ? game.involved_companies[1].company.name : game.involved_companies[0].company.name} </div> :
                                " "
                            }
                          </Grid>
                          <Grid item className={styles.hm}>
                            <div className={styles.rate}> {game.summary ? game.summary : game.name + " video game"} </div>
                          </Grid>

                        </Grid>
                      </Grid>

                      <Grid item xs={12} md={4} lg={3}>
                        <div className={"aggregated_rating" in game ? game.aggregated_rating < 70 ? game.aggregated_rating < 50 ? `${styles.gamerating} ${styles.bad}`: `${styles.gamerating} ${styles.okay}`: `${styles.gamerating} ${styles.good}`: `${styles.gamerating} ${styles.tbd}`}> { ('aggregated_rating' in game)? parseInt(game.aggregated_rating ) : "tbd" } </div>

                      </Grid>
                    </Grid>
                  </div>
                  <div className="styles.screens">
                    <GridList className={styles.gridList} cols={2.3} cellHeight={300}>
                      {game.screenshots ? game.screenshots.map(screen => (
                        <GridListTile key={screen.image_id}>
                          <img src={"https://images.igdb.com/igdb/image/upload/t_screenshot_med/" + screen.image_id + ".jpg"} alt={screen.image_id}/>
                        </GridListTile>
                      )) : " "}
                    </GridList>
                </div>
              </Grid>
            </Grid>
        )
    }
}
