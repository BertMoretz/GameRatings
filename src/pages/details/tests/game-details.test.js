import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import createStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import {GameDetails} from '../game-details'

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import { shallow, mount, render } from 'enzyme';

describe('GameDetails', () => {
  it('test render empty component', () => {
    const initialState = {
      game: void 0
    }

    const container = renderer
      .create(
        <Router>
          <Route component={GameDetails}/>
        </Router>
      )

      expect(container.toJSON()).toMatchSnapshot()
    })
})
