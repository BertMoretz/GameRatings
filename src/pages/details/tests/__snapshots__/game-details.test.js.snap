// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameDetails test render empty component 1`] = `null`;
